### ISSUE TITLE
Deprecate thrift and memcached transports

### ISSUE THREAD

**Issue by clintongormley** [elastic/elasticsearch#10166]
The thrift and memcached transport layers were experimental, but in the end didn't buy us much.  Memcached is very limited in scope, supporting only a portion of the REST API, and thrift has much the same throughput as HTTP.

Deprecate in 1.5, and remove in 2.0

**Comment by mrsolo** [comment]
http://build-eu-00.elastic.co/view/Clients/job/es-py_core/ thirft/memcached disabled

**Comment by piotrjurkiewicz** [comment]
I use memcached transport in production for storing log messages and I am able to achieve far more higher throughput than with HTTP REST API.

Furthermore, memcached protocol can work in connectionless mode. This plugin can be fairly simple extended to allow binding to UDP and DGRAM sockets. Memcached client libraries already support that. This would make possible to use ES in connectionless mode. This will suit perfectly for storing log messages.

Do ES team have another ideas how to provide connectionless operation?

**Comment by q42mark** [comment]
In theory TCP+thrift should provide better performance. What is the reason to drop this? Could you give us a more detailed explanation of why this is removed?

**Comment by trentcioran** [comment]
yes, could you provide more detailed explanation? Thanks

**Comment by micherr** [comment]
We've been investigating for a while and just decided to switch to thrift for transporting. 
Do you plan any alternative in 2.0 in case of depreciation?

**Comment by xenji** [comment]
Same thing with us. We've decided to build a new part of the architecture using thrift as default communication technology. Although we are not doomed when ES removes thrift, it is still a desirable piece in the puzzle. The Thrift TSocket communication is also much faster than Thrift over http (about 5 to 10 times in our tests).

Have you taken the thrift socket communication option into consideration while benchmarking?

**Comment by wkv** [comment]
The Thrift interface is 5 to 10 times faster than the HTTP interface and has been for 3 years now.  What testing has shown otherwise?

**Comment by 3lnc** [comment]
Same question about thrift. Any benchmarks about it? Any potential replacement so far?

**Comment by clintongormley** [comment]
Hi all

We have compared HTTP to Thrift on the officially supported clients in .net, python, ruby, and perl, (with fast http backends, persistent connections, good json encoders etc) and we all found that the performance difference was negligible.  This is why we decided to stop supporting thrift.  I'd be interested to see the code that you used where you found such large performance differences.
